packages/cli/src/dbt/schema.ts

```diff
@@ -35,3 +35,54 @@ export const loadYamlSchema = async (path: string): Promise<YamlSchema> => {
     }
     return schemaFile;
 };
+
+type FindModelInYamlArgs = {
+    filename: string;
+    modelName: string;
+};
+const findModelInYaml = async ({
+    filename,
+    modelName,
+}: FindModelInYamlArgs) => {
+    try {
+        const existingYaml = await loadYamlSchema(filename);
+        const models = existingYaml.models || [];
+        const modelIndex = models.findIndex(
+            (model) => model.name === modelName,
+        );
+        if (modelIndex < 0) {
+            return undefined;
+        }
+        return {
+            doc: existingYaml as YamlSchema &
+                Required<Pick<YamlSchema, 'models'>>,
+            modelIndex,
+        };
+    } catch (e) {
+        if (e.code === 'ENOENT') {
+            return undefined;
+        }
+        throw e;
+    }
+};
+
+type SearchForModelArgs = {
+    modelName: string;
+    filenames: string[];
+};
+export const searchForModel = async ({
+    modelName,
+    filenames,
+}: SearchForModelArgs) => {
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
why is this throwing a eslint error ?

Reply:
_sighs_ I don't know why it's default: https://eslint.org/docs/rules/no-restricted-syntax

We can configure that, should I update eslint?

Reply:
Ah it's a hotly debated feature in the airbnb rules we use: https://github.com/airbnb/javascript/issues/1271

I'm going to overwrite it, I'm personally happy to use for loops (since code-readability for `await`ed loops is 100x better than any alternative)